### PR TITLE
Help Up instead of Shake Up

### PR DESCRIPTION
**Pull request by Mwahahahaha**
Whenever you use help intent on someone who is lying down, you hold out your hand instead.
The person lying down can then click on you with help intent to take that hand and be helped up.

I also ported over Polaris' code for hoisting yourself up by grabbing onto other people but sets them on fire if you're on fire. I also modified the time it takes to pull yourself up so it's longer than in Polaris.

Clicking on the person lying down again, moving away, or getting incapacitated will retract your hand.

Clicking on the person helping you up while lying down with harm intent slaps away their hand.

Don't be a brute.
Teamwork makes the dream work.

[Alt text](https://youtu.be/TgAtfjRg72w)

### CODE REVIEW

**Comment by Mwahahahaha** [comment]
Okay so like
TIL you can still punch and grab people from the ground which is pretty sick — I love that.
But like,

image

I don't really know what to do with the intense hand-slapping action now.
Should I keep it anyways or do I remove it?

**Comment by Mwahahahaha** [comment]
!review

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -288,15 +290,40 @@
...
-			else if(lying || src.sleeping)
-				src.sleeping = max(0,src.sleeping-5)
-				if(src.sleeping == 0)
-					src.resting = 0
-				M.visible_message("<span class='notice'>[M] shakes [src] trying to wake [t_him] up!</span>", \
-									"<span class='notice'>You shake [src] trying to wake [t_him] up!</span>")
+			else if(lying)
+				if(src.sleeping)
+					src.sleeping = max(0,src.sleeping-5)
+					M.visible_message("<span class='notice'>[M] shakes [src] trying to wake [t_him] up!</span>", \
```
Comment (alsoandanswer) [inline code/modules/mob/living/carbon/carbon.dm]:
```suggestion
					M.visible_message(span("notice", "[M] shakes [src] trying to wake [t_him] up!"), \
```

Change span class notice to span proc, thanks

File: code/modules/mob/living/carbon/carbon.dm [hunk 2]
```diff
@@ -288,15 +290,40 @@
...
+										"<span class='notice'>You shake [src] trying to wake [t_him] up!</span>")
```
Comment (alsoandanswer) [inline code/modules/mob/living/carbon/carbon.dm]:
```suggestion
										span("notice", "You shake [src] trying to wake [t_him] up!"))
```

**Comment by alsoandanswer** [comment]
also welcome back

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -287,17 +289,42 @@
...
+					M.visible_message(span("notice", "[M] shakes [src] trying to wake [t_him] up!"), \
+										span("notice", "You shake [src] trying to wake [t_him] up!"))
+				else
+					M.help_up_offer = !M.help_up_offer
+					if(M.help_up_offer)
+						M.visible_message(span("notice", "[M] holds a hand out to [src]."), \
+											span("notice", "You hold a hand out to [src]."))
+					else
+						M.visible_message(span("warning", "[M] retracts their hand from [src]'s direction."), \
+											span("warning", "You retract your hand from [src]'s direction."))
```
Comment (skull132) [inline code/modules/mob/living/carbon/carbon.dm]: The fact that this uses the M mob's variable is odd. What happens when 2 people offer the same person help? Basically, first one will get `You hold a hand out to [src].` and the other one will get `You retract your hand from [src]'s direction.`, despite the other one not having offered help in the first place.

It should either use the `src` variable of it, or store a reference to the mob offering help.

Comment (Mwahahahaha) [reply code/modules/mob/living/carbon/carbon.dm]: image

I'm not really sure this seems to be case, though I apologise if I may have misunderstood what you're pointing out.

What DOES happen, though, is that if the person on the ground takes one of the people's hands, the other person remains holding their hand out. This is reset to 0 though as soon as that other person does any action to automatically reset it to 0, such as by moving.

**Pull request merged**